### PR TITLE
Generate webauthn verification OTP after authentication

### PR DESCRIPTION
**Pull request by jenshenny**
## What problem are you solving?

Part of https://github.com/rubygems/rubygems.org/pull/3298

After successful verification of Webauthn credentials, an OTP needs generated to be sent to the client (either through manual pasting or sockets).

## What approach did you choose and why?
- Added a generate otp method on WebauthnVerification model
- After webauthn credentials are verified, generate the otp as `SecureRandom.base58(20)` with 2 minute expiry.

## What should reviewers focus on?
- is `SecureRandom.base58(20)` fine for the otp and 2 minutes as the expiry?

## Testing
1. Follow the tophat instructions from https://github.com/rubygems/rubygems.org/pull/3324 to get to the prompt page. You are able to authenticate your webauthn credential when clicking the button.
2. Check if an OTP is generated by going to the console and running `WebauthnVerification.find_by(path_token: <path_token>).otp`

### CODE REVIEW

**Comment by jenshenny** [comment]
cc: @jchestershopify @bettymakes

File: test/unit/webauthn_verification_test.rb [hunk 1]
```diff
@@ -50,4 +50,23 @@ class WebauthnVerificationTest < ActiveSupport::TestCase
...
+      @generated_time = Time.utc(2023, 1, 1, 0, 0, 0)
+      travel_to @generated_time do
+        @webauthn_verification.generate_otp
+      end
+      @webauthn_verification.reload
+    end
+
+    should "create a token that is 20 characters long" do
+      assert_equal 20, @webauthn_verification.otp.length
+    end
```
Comment (jchestershopify) [inline test/unit/webauthn_verification_test.rb:66]: IIRC we're using 16 characters for the path_token. Should we use the same here?

**Review by jchestershopify — COMMENTED** [review]
One question about token size, but otherwise LGTM.

**Review by aellispierce — APPROVED** [review]
This LGTM 🎉 . I tophatted locally and got the OTP code as expected

File: test/unit/webauthn_verification_test.rb (hunk 1 above)
Comment (jenshenny) [reply test/unit/webauthn_verification_test.rb]: Good point, I was basing the number off of the number set in the factory which was 20. I like the consistency of 16 so I changed it to that ✅

Comment (jchestershopify) [reply test/unit/webauthn_verification_test.rb]: Aha! I thought I'd changed that in #3328, but I think it got lost in the general rebase shuffle.

**Pull request merged**